dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/RelationAnalysis.java

```diff
@@ -38,14 +89,24 @@ private RelationAnalysis(VerificationTask task, Context context, Configuration c
      *                    <li>{@link AliasAnalysis}
      *                    <li>{@link WmmAnalysis}
      *                </ul>
-     * @param config User-defined options to further specify the behavior.
+     * @param config  User-defined options to further specify the behavior.
      */
     public static RelationAnalysis fromConfig(VerificationTask task, Context context, Configuration config) throws InvalidConfigurationException {
         RelationAnalysis a = new RelationAnalysis(task, context, config);
-        a.run(task, context);
+        a.run(task.getMemoryModel());
         return a;
     }
 
+    /**
+     * Fetches results of this analysis.
+     *
+     * @param relation Some element in the associated task's memory model.
+     * @return Local fraction of results.
```

Comment:
"Computed knowledge of relation" or something like that would be better.

Reply:
Your suggestion is also good.

In general though, I don't think that the java docs necessarily need to say more than the function name + signature itself (unless there is more to tell).
The reason is simply because well-designed function names may already tell the "whole story".
In that case, changing the docs just to avoid redundancy is not benefitial IMO.
I guess this is also a reason why I avoid writing docs and instead make the function itself more descriptive, if possible.